ocaml/typing/typetexp.ml

```diff
@@ -677,6 +677,20 @@ and transl_type_aux env policy mode styp =
       in
       List.iter2
         (fun (sty, cty) ty' ->
+           let get_reason layout = match Layout.get layout with
+             | Const Value -> Layout.(Value_creation (Imported_type_argument path))
+             (* CR layouts: Add more cases here when type params of imported types can
```

Comment:
This is already true: we can import definitions with non-value layout arguments.

But I think this doesn't need to match on the layout at all. Instead of putting `Imported_type_argument` into `value_creation_reason`, put it elsewhere to avoid the need to worry about which layout the argument has.

Reply:
Added `Imported_type_argument` as just a `creation_reason` to eliminate the need for the match.

This makes `Imported_type_argument` and `Type_argument` different again (one is a `creation_reason` and the other is a `value_creation_reason`).